test/health_cards/health_card_test.rb

```diff
@@ -95,44 +99,66 @@ class HealthCardTest < ActiveSupport::TestCase
     assert_equal(resource_nums, inc_array)
   end
 
+  test 'changes to stripped bundle do not affect bundle values' do
+    stripped_bundle = @health_card.strip_fhir_bundle
+
+    # Check bundle
+    stripped_bundle.type = 'foobar'
+    assert_not_equal @health_card.bundle.type, stripped_bundle.type
+
+    # Check entry value
+    patient = stripped_bundle.entry[0].resource
+    patient.name[0].family = 'foobar'
+    assert_not_equal @health_card.bundle.entry[0].resource.name[0].family,
+                     stripped_bundle.entry[0].resource.name[0].family
+
+    # Check entries
+    stripped_bundle.entry = []
+    assert_not_equal @health_card.bundle.entry.length, stripped_bundle.entry.length
+  end
+
   test 'update_elements strips resource-level "id", "meta", and "text" elements from the FHIR Bundle' do
     stripped_bundle = @health_card.strip_fhir_bundle
-    stripped_entries = stripped_bundle['entry']
+    stripped_entries = stripped_bundle.entry
 
     stripped_entries.each do |entry|
-      resource = entry['resource']
-      assert_not(resource.key?('id'))
-      assert_not(resource.key?('text'))
-      assert(!resource.key?('meta') || (resource.key?('meta') && (resource['meta'].keys == ['security'])))
+      resource = entry.resource
+      assert_not(resource.id)
+      assert_not(resource.text)
+      meta = resource.meta
+      if meta
+        assert_equal 1, meta.to_hash.length
+        assert_not_nil meta.security
+      end
     end
   end
 
   test 'update_nested_elements strips any "CodeableConcept.text" and "Coding.display" elements from the FHIR Bundle' do
     stripped_bundle = @health_card.strip_fhir_bundle
-    stripped_resources = stripped_bundle['entry']
+    stripped_resources = stripped_bundle.entry
 
     resource_with_codeable_concept = stripped_resources[2]
-    codeable_concept = resource_with_codeable_concept['resource']['valueCodeableConcept']
-    coding = codeable_concept['coding'][0]
+    codeable_concept = resource_with_codeable_concept.resource.valueCodeableConcept
+    coding = codeable_concept.coding[0]
 
-    assert_not(codeable_concept.key?('text'))
-    assert_not(coding.key?('display'))
+    assert_nil codeable_concept.text
+    assert_nil coding.display
   end
 
   test 'update_nested_elements populates Reference.reference elements with short resource-scheme URIs' do
     stripped_bundle = @health_card.strip_fhir_bundle
-    stripped_resources = stripped_bundle['entry']
+    stripped_resources = stripped_bundle.entry
     resource_with_reference = stripped_resources[2]
 
-    reference = resource_with_reference['resource']['subject']['reference']
+    reference = resource_with_reference.resource.subject.reference
     assert(reference.start_with?('resource:') && (reference.length <= 12))
```

Comment:
This is just a question, but do you know if there is a definition somewhere for "short resource-scheme URIs"? I think the motivation for checking if length was less than 12 was that it can contain "resource:" + a 2 digit number, but I'm realizing now that this test is somewhat arbitrary.

Reply:
I don't see it in the spec anywhere, not sure if @masnick or @radamson know. We could change the test to use a regex instead of the existing assertion.

Reply:
I changed this test to use a regex but it would be good to know either way if there's any other rules around this url

Reply:
Let's bring it up at standup next week.